app/src/main/java/com/weatherxm/util/Weather.kt

```diff
@@ -106,15 +106,13 @@ object Weather : KoinComponent {
             return "$EMPTY_VALUE$unit"
         }
 
-        return if (ignoreConversion) {
-            if (decimals == 0) {
-                "${roundToInt(value)}$unit"
-            } else {
-                "${roundToDecimals(value)}$unit"
-            }
+        val formattedValue = if (ignoreConversion) {
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/WeatherXM/wxm-android/pull/342#issuecomment-2391143597>#b921d3</a><span> - 10/03/2024, 11:34 am</span>
<br>

<div><small>🟠</small> Medium importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The function uses multiple boolean parameters, which can make the code less readable and more prone to errors when calling the function.
</div>
<div id="fix">
<b>Fix:</b> Consider using a data class or a builder pattern to replace the multiple boolean parameters. This will make the function calls more readable and less error-prone.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -109,13 +109,26 @@ object Weather : KoinComponent {
 -    fun getFormattedTemperature(
 -        value: Float?,
 -        decimals: Int = 0,
 -        fullUnit: Boolean = true,
 -        includeUnit: Boolean = true,
 -        ignoreConversion: Boolean = false
 -    ): String {
 -        val formattedValue = if (ignoreConversion) {
 -            formatNumber(value, decimals)
 -        } else {
 -            formatNumber(convertTemp(value, decimals), decimals)
 -        }
 -
 -        return "$formattedValue$unit"
 +    data class TemperatureFormatOptions(
 +        val decimals: Int = 0,
 +        val fullUnit: Boolean = true,
 +        val includeUnit: Boolean = true,
 +        val ignoreConversion: Boolean = false
 +    )
 +
 +    fun getFormattedTemperature(
 +        value: Float?,
 +        options: TemperatureFormatOptions = TemperatureFormatOptions()
 +    ): String {
 +        val (decimals, fullUnit, includeUnit, ignoreConversion) = options
 +        val formattedValue = if (ignoreConversion) {
 +            formatNumber(value, decimals)
 +        } else {
 +            formatNumber(convertTemp(value, decimals), decimals)
 +        }
 +
 +        val unit = if (includeUnit) {
 +            if (fullUnit) "°${getTemperatureUnit()}" else "°"
 +        } else ""
 +
 +        return "$formattedValue$unit"
 +    }
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged